xchange-stream-kraken/src/main/java/info/bitrich/xchangestream/kraken/KrakenStreamingAdapters.java

```diff
@@ -90,7 +91,8 @@ public static OrderBook adaptOrderbookMessage(int depth, TreeSet<LimitOrder> bid
         } else if (bids.size() > 0 && asks.size() > 0 && bids.first().getLimitPrice().compareTo(asks.first().getLimitPrice()) >= 0) {
             throw new IllegalStateException("CROSSED book " + instrument + " " + bids.first().getLimitPrice() + " >= " + asks.first().getLimitPrice());
         }
-        return new OrderBook(lastTime.get(), Lists.newArrayList(asks), Lists.newArrayList(bids), true);
+        final Date lastTime = Stream.concat(asks.stream(), bids.stream()).map(LimitOrder::getTimestamp).max(Date::compareTo).orElse(Date.from(Instant.EPOCH));
```

Comment:
the date should be null if we cannot figure out the data from the snapshot or bids/asks because otherwise you are just timestamping the time we received the message

Reply:
fixed

Reply:
fixed